src/pip/_internal/req/req_install.py

```diff
@@ -633,6 +634,18 @@ def update_editable(self, obtain=True):
         vc_type, url = self.link.url.split('+', 1)
         vcs_backend = vcs.get_backend(vc_type)
         if vcs_backend:
+            if not self.link.is_vcs:
+                reason = (
+                    "This form of VCS requirement is being deprecated: {}."
+                ).format(
+                    self.link.url
+                )
+                replacement = None
+                if self.link.url.startswith("git+git@"):
+                    replacement = (
+                        "git+https:// or git+ssh://"
+                    )
+                deprecated(reason, replacement, gone_in="21.0")
```

Comment:
I would make an issue describing the deprecation decision, close #2994 and #6293 against it, and then put it here like `issue=XXXX`. That will let us capture feedback in one place and can also represent the cleanup work that is necessary once the deprecation period is over. See #7297 as an example.

Reply:
I created the tracking issue (#7554) and made the replacement message more explicit.